Plutarch/DataRepr/Internal.hs

```diff
@@ -121,6 +124,35 @@ instance PlutusType (PDataRecord '[]) where
   pcon' PDNil = pnil
   pmatch' _ f = f PDNil
 
+-- | This uses data equality. 'PEq' instances of elements don't make any difference.
+instance PEq (PDataRecord xs) where
+  x #== y = pdToBuiltin x #== pdToBuiltin y
+
+-- Lexicographic ordering based 'Ord' instances for 'PDataRecord'.
+
+instance POrd (PDataRecord '[]) where
+  _ #<= _ = pconstant True
+  _ #< _ = pconstant False
+
+instance (POrd x, PIsData x) => POrd (PDataRecord '[label ':= x]) where
+  l1 #< l2 = unTermCont $ do
```

Comment:
What if you use `#<` on the same type a lot?

Reply:
there's a delicate balance here that I'm not totally sure how to handle. I think hoisting should be done at the `PDataSum` level since there won't be any recursion there. If I hoist it for `PDataRecord`s, for every single field a new `plam` would be introduced. Right now, it all just nicely unrolls into a bunch of field comparisons - as if you wrote it by hand.

At the same time, hoisting for larger `PDataRecord`s (say, 5 elements? yet to determine) might actually be good - it could potentially have a sort of "memoization" effect, where multiple comparisons can reference one function that compares 5 fields at once (no inner `plam`s for each field). This sort of batch hoisting seems good in my intuition - but all of this needs to be tested.

Reply:
decided to use `phoistAcyclic` at the `PDataSum` level - it should be optimal for multi-use cases. Slightly worse for single worse case due to the lambda application